### PR TITLE
Add response HTTP status to Networking engine responses

### PR DESCRIPTION
**Pull request by Pavelas**
## Description
This pull request fixes the issue: #3640 

Basically I just added response status to response filter, it can be useful for someone who needs to check response status and based on that do some actions.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have signed the Google CLA <https://cla.developers.google.com>
- [x] My code follows the style guidelines of this project
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have verified my change on multiple browsers on different platforms
- [x] I have run `./build/all.py` and the build passes
- [x] I have run `./build/test.py` and all tests pass

### CODE REVIEW

File: externs/shaka/net.js [hunk 1]
```diff
@@ -118,6 +118,8 @@ shaka.extern.Request;
  *   The original URI passed to the browser for networking. This is before any
  *   redirects, but after request filters are executed.
  * @property {BufferSource} data
+ *   The response HTTP status code.
```
Comment (avelad) [inline externs/shaka/net.js:121]: I think the comment is inverted, this should go in status and the status in data.

File: externs/shaka/net.js [hunk 2]
```diff
@@ -118,6 +118,8 @@ shaka.extern.Request;
...
+  * @property {number} status
```
Comment (avelad) [inline externs/shaka/net.js:122]: Should be ` * @property {(number|undefined)} status`  In the offline plugin it can be optional.

Comment (avelad) [reply externs/shaka/net.js:122]: I think you are missing a part in the definition, you should add `* status: (number | undefined)` in line 107

File: externs/shaka/net.js (hunk 1 above)
Comment (Pavelas) [reply externs/shaka/net.js]: Omg! I am so sorry for this, fixed!

File: externs/shaka/net.js (hunk 2 above)
Comment (Pavelas) [reply externs/shaka/net.js]: Thank you very much for your comments, I was missing this!

**Comment by Pavelas** [comment]
@googlebot I fixed it.

**Review by theodab — APPROVED** [review]

File: externs/shaka/net.js [hunk 3]
```diff
@@ -101,6 +101,7 @@ shaka.extern.Request;
  * @typedef {{
  *   uri: string,
  *   data: BufferSource,
+ *   status: (number|undefined)
```
Comment (theodab) [inline externs/shaka/net.js:104]: Oh! You forgot to put a comma at the end of this line. Thanks, build bot.

Comment (Pavelas) [reply externs/shaka/net.js]: Stupid me! Sorry! Definitely lack of sleep in the past days...

**Comment by theodab** [comment]
Thanks for your contributions!
Also, thanks for the code review, Alvaro.

**Pull request merged**